### PR TITLE
Updated msp_merge_ancestors to match Python implementation

### PR DESCRIPTION
**Pull request by nikbaya**

### CODE REVIEW

File: lib/msprime.c [hunk 1]
```diff
@@ -3169,15 +3169,15 @@ msp_merge_ancestors(msp_t *self, avl_tree_t *Q, population_id_t population_id,
...
+            if (new_node_id == TSK_NULL) {
                 coalescence = true;
                 l_min = l;
                 ret = msp_store_node(self, 0, self->time, population_id, individual);
                 if (ret != 0) {
                     goto out;
                 }
             }
-            v = (tsk_id_t) msp_get_num_nodes(self) - 1;
+            new_node_id = (tsk_id_t) msp_get_num_nodes(self) - 1;
```
Comment (jeromekelleher) [inline lib/msprime.c:3180]: This needs to be *inside* the if statement, otherwise we overwrite the supplied new node id value.

**Review by jeromekelleher — CHANGES_REQUESTED** [review]
Looks good, but there's a logic issue.

**Comment by jeromekelleher** [comment]
Yep, that's it! Now, we just need to [squash](https://stdpopsim.readthedocs.io/en/latest/development.html#rebasing) the commits into done, and we're good to merge.

**Comment by nikbaya** [comment]
@jeromekelleher I think I squashed the commits, is there more to do before completing the PR?

**Review by jeromekelleher — APPROVED** [review]
All done, thanks @nikbaya!

**Pull request merged**